src/devices/GoPiGo3/GoPiGo3.cs

```diff
@@ -0,0 +1,900 @@
+            else
+            {
+                throw new IOException("No SPI response");
+            }
+            return retVal;
+        }
+
+        /// <summary>
+        /// Read the firmware version
+        /// </summary>
+        /// <returns>Returns the firmware version</returns>
+        public string GetFirmwareVersion()
+        {
+            string retVal = string.Empty;
+            int version = SpiRead32(SpiMessageType.GetFirmwareVersion);
+            retVal = string.Format("{0}.{1}.{2}", (version / 1000000), ((version / 1000) % 1000), (version % 1000));
+            return retVal;
+        }
+
+        /// <summary>
+        /// Set the led color
+        /// </summary>
+        /// <param name="led">The led, either Left, Right, Blinky Left, Blinky Right and Wifi. Note you should only control the wifi one if you are sure to be connected to wifi</param>
+        /// <param name="red">The red component</param>
+        /// <param name="green">The green component</param>
+        /// <param name="blue">The blue component</param>
+        public void SetLed(byte led, byte red, byte green = 0, byte blue = 0)
+        {
+
+            byte[] outArray = { SpiAddress, (byte)SpiMessageType.SetLed, led, red, green, blue };
+            var reply = SpiTransferArray(outArray);
+        }
+
+        /// <summary>
+        /// Get the 5v circuit voltage
+        /// </summary>
+        /// <returns>5v circuit voltage</returns>
+        public double GetVoltage5V()
+        {
+            var value = SpiRead16(SpiMessageType.GetVoltage5V);
+            return (value / 1000.0);
+        }
+
+        /// <summary>
+        /// Get the battery voltage
+        /// </summary>
+        /// <returns>battery/Vcc voltage</returns>
+        public double GetVoltageBatteryVcc()
+        {
+            var value = SpiRead16(SpiMessageType.GetVoltageVcc);
+            return (value / 1000.0);
+        }
+
+        #endregion
+
+        #region Servo and motors
+
+        /// <summary>
+        /// Move the servo motor to a specific pulse
+        /// </summary>
+        /// <param name="servo">Tjhe servo port Servo1 or Servo2</param>
+        /// <param name="pulseMicroseconds">The pulse in microseconds</param>
+        public void SetServo(ServoPort servo, int pulseMicroseconds)
+        {
+            byte[] outArray = { SpiAddress, (byte)SpiMessageType.SetServo, (byte)servo, (byte)((pulseMicroseconds >> 8) & 0xFF), (byte)(pulseMicroseconds & 0xFF) };
+            var reply = SpiTransferArray(outArray);
+        }
+
+        /// <summary>
+        /// Set the motor power in percent
+        /// </summary>
+        /// <param name="port">The Motor port to use, can be MotorLeft and/or MotorRight</param>
+        /// <param name="powerPercent">The power from - 100 to 100, or -128 for float</param>
+        public void SetMotorPower(MotorPort port, int powerPercent)
+        {
+            powerPercent = powerPercent > 127 ? (byte)127 : powerPercent;
```

Comment:
why is percentage from -127 to 127?

Reply:
The power is from -100 to 100, anything else just float the motor. I renamed